charms/k8s/src/charm.py

```diff
@@ -16,35 +16,143 @@
 """
 
 import logging
+import re
+import shlex
+import subprocess
+from typing import Optional
 
 import ops
+from charms.k8s.v0.k8sd_api_manager import (
+    InvalidResponseError,
+    K8sdAPIManager,
+    K8sdConnectionError,
+    UnixSocketConnectionFactory,
+)
+from charms.operator_libs_linux.v2.snap import SnapCache, SnapError, SnapState
+from ops.model import WaitingStatus
+from pydantic import ValidationError
 
 # Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
 VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
-K8SD_SNAP_SOCKET = "/var/snap/k8s/common/var/lib/k8sd/control.socket"
+K8SD_SNAP_SOCKET = "/var/snap/k8s/common/var/lib/k8sd/state/control.socket"
 
 
 class K8sCharm(ops.CharmBase):
-    """Charm the service."""
+    """A charm for managing a K8s cluster via the k8s snap."""
 
     def __init__(self, *args):
-        """Construct.
+        """Initialise the K8s charm.
 
         Args:
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
         self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.install, self._on_install)
 
-    def _on_update_status(self, _event: ops.UpdateStatusEvent):
+        factory = UnixSocketConnectionFactory(unix_socket=K8SD_SNAP_SOCKET)
+        self.api_manager = K8sdAPIManager(factory)
+        self.snap_cache = SnapCache()
+
+    def _apply_snap_requirements(self):
+        """Apply necessary snap requirements for the k8s snap.
+
+        This method executes necessary scripts to ensure that the snap
+        meets the network and interface requirements.
+        """
+        self.unit.status = ops.MaintenanceStatus("Applying K8s requirements")
+        commands = [
+            "/snap/k8s/current/k8s/connect-interfaces.sh",
```

Comment:
This is only needed when we deploy a local built of the snap.

Reply:
It sounds like we are relying on `connect-interfaces.sh` to enable some interfaces in the snap. If I remove the call, the charm does not become active.